### PR TITLE
add Azure base map & handle layer group

### PR DESCRIPTION
**Pull request by Sonia-96**
### Description

- Support Azure Maps in Gizmo Map View
- Use ol.layer.Group to create an array of basemap layers

### Changes Made to Code:
 - Create a function `ol_base_map_layer_init` to create a new base map layer, which can be either a single layer or a layer group

### Quality Checks
 - [] New code is 100% tested
 - [x] Code has been formated
 - [x] Code has been linted
 - [x] Docstrings for new methods have been added

### CODE REVIEW

File: tethys_gizmos/static/tethys_gizmos/js/tethys_map_view.js [hunk 1]
```diff
@@ -7,1861 +7,2165 @@
...
-        if (typeof base_map_option === 'string') {
-            base_map_layer_name = base_map_option;
-            base_map_layer_arguments = null;
-        }
-        else if (typeof base_map_option === 'object'){
-            base_map_layer_name = Object.getOwnPropertyNames(base_map_option)[0];
-            base_map_layer_arguments = base_map_option[base_map_layer_name];
+      if (typeof base_map_option === 'string') {
+          base_map_layer_names.push(base_map_option);
+          base_map_layer_arguments = null;
```
Comment (swainn) [inline tethys_gizmos/static/tethys_gizmos/js/tethys_map_view.js:308]: Should this be?

 ```
basemap_layer_arguments.push(null);
```

File: tethys_gizmos/static/tethys_gizmos/js/tethys_map_view.js [hunk 2]
```diff
@@ -7,1861 +7,2165 @@
...
+  // Constants
+  var SUPPORTED_BASE_MAPS = {
+  'OpenStreetMap': {
+      source_class: ol.source.OSM,
+      default_source_options: {},
+      label_property: null,
+  },
+  'Bing': {
+      source_class: ol.source.BingMaps,
+      default_source_options: null,
```
Comment (swainn) [inline tethys_gizmos/static/tethys_gizmos/js/tethys_map_view.js:178]: Please also add a Deprecation warning to the MapView gizmo constructor [when "Bing" is included in the `basemap` argument](https://github.com/tethysplatform/tethys/blob/main/tethys_gizmos/gizmo_options/map_view.py#L352). Use the `deprectation_warning` function provided by Tethys. [Here is an example of it's usage.](https://github.com/tethysplatform/tethys/blob/8680dcfc8595c30bf4e47be110d7f0fb5d1b9f60/tethys_cli/manage_commands.py#L123-L131)

File: tethys_gizmos/gizmo_options/map_view.py [hunk 3]
```diff
@@ -57,15 +58,21 @@ class MapView(TethysGizmoOptions):
...
+    There are several base maps supported by the Map View gizmo: `OpenStreetMap`, `Bing`, `Azure`, `CartoDB`, and `ESRI`. All base maps can be specified as a string or as an options dictionary. When using an options dictionary all base maps map services accept the option `control_label`, which is used to specify the label to be used in the Base Map control. For example::
 
         {'Bing': {'key': 'Ap|k3yheRE', 'imagerySet': 'Aerial', 'control_label': 'Bing Aerial'}}
+        {'Azure': {'tilesetId': 'microsoft.imagery', 'subscriptionKey': 'Ap|k3yheRE', 'layer': 'Imagery'}}
+
+    .. note::
+
+        The Bing Map service has been depricated in favor of Azure Maps service.
+        For moe options for Azure Maps tilesetId, please refer to the link: https://learn.microsoft.com/en-us/rest/api/maps/render/get-map-tile?view=rest-maps-2024-04-01&tabs=HTTP#tilesetid
```
Comment (swainn) [inline tethys_gizmos/gizmo_options/map_view.py:69]:
```suggestion
        For more options for Azure Maps tilesetId, please refer to this link: `TilesetID <https://learn.microsoft.com/en-us/rest/api/maps/render/get-map-tile?view=rest-maps-2024-04-01&tabs=HTTP#tilesetid>`_
```

File: tethys_gizmos/gizmo_options/map_view.py [hunk 4]
```diff
@@ -349,6 +361,29 @@ def __init__(
...
+
+        def check_bing_map(map_type):
+            if map_type.lower() == 'bing':
+                deprecation_warning(
+                    version="5.0",
+                    feature='the Bing base map',
+                    message="The Bing Map service has been depricated in favor of Azure Maps service and will be "
+                    'retired on June 30, 2028. Please switch to Azure Maps at your erliest convenience. '
+                    f"For instructions on using Azure Maps, see "
+                    f"{DOCS_BASE_URL}/tethys_sdk/gizmos/map_view.html",
```
Comment (swainn) [inline tethys_gizmos/gizmo_options/map_view.py:373]:
```suggestion
                    message="The Bing Map service has been deprecated in favor of Azure Maps service and will be "
                    'retired on June 30, 2028. Please switch to Azure Maps at your earliest convenience. '
                    f"For instructions on using Azure Maps, see "
                    f"{DOCS_BASE_URL}/tethys_sdk/gizmos/map_view.html",
```

**Review by swainn — APPROVED** [review]

**Review by shawncrawley — APPROVED** [review]
There were a lot of changes there in `tethys_map_view.js`! It looks good to me, though. I approve.

**Pull request merged**